### PR TITLE
Pydantic dataclasses

### LINKED ISSUE #63: Image parser: Use datacache
The current image parse strategy takes an image file and creates a cropped file with the same name as the original file, with the prefix "cropped_" added. Both files are stored as local files, but we generally want our resources stored in the datacache instead.

The parser should therefore take a datacache key as input to `get()`, store the cropped file in the datacache, and return the datacache key to the cropped image, like e.g. the download strategies do.

### LINKED ISSUE #68: Remove image/eps as supported image format
The image parse strategy is based on the `Image` class in the `Pillow` package, which supports the `EPS (Encapsulated PostScript)` format. However, reading an `EPS` image requires that `Ghostscript` is installed on the system.

Instead of requiring `Ghostscript` to be installed, we should remove `image/eps` from the list of supported image formats.

### LINKED ISSUE #74: Use proper file scheme URLs in tests
#35 introduced some misuse of `file`-scheme URLs.
A correct `file` scheme url is of the form:
`file:///path/to/file` or `file://hostname/path/to/file`
Where, if `hostname` is `localhost`, the two examples above are equivalent.
The tests use `file://path/to/file`, missing a forward slash and incorrectly using the `host` part of the parsed URL in the `FileStrategy`.

The correct solution here would be to check whether `host` is `None` and use it if not, and otherwise ignore it. This ensures a full handling of `file`-scheme URLs.

### LINKED ISSUE #75: Clean up tests
Static test files, e.g. used for testing strategies, should be placed under the `tests/static/` folder.
Static typing should be fully implemented for all tests.

### PR DESCRIPTION
**Pull request by CasperWA**
Fixes #74 
Closes #75 
Closes #68 
Closes #63 

Use `pydantic.dataclasses.dataclass` instead of `dataclasses.dataclass`.
This ensures the `*_config` attribute is type-checked, meaning one can now sub-class the `*Config` pydantic model and substitute this in for the specific strategy, updating the `configuration` field with another strategy-specific model that then automatically will ensure the configuration for the strategy is correct upon strategy initialization.

Mypy has been extended with the pydantic plugin.

The tests have been slightly streamlined.

This PR also implements the beginnings of satisfying #71. However, it doesn't implement an `introspection` or `schema` method to fully satisfy #71.

### CODE REVIEW

**Comment by CasperWA** [comment]
> Will this PR influences OTEAPI-services and OTELIB?

I don't think it should. Everything should be fine from their perspective. But it's difficult to say 100 %. Should be fine though :)

**Comment by CasperWA** [comment]
@jesper-friis I have now updated this PR with what we discussed offline concerning having a utility function for converting URI/IRI/URLs to `pathlib.Path`.

File: oteapi/strategies/download/sftp.py [hunk 1]
```diff
@@ -1,21 +1,19 @@
 """Strategy class for sftp/ftp"""
 # pylint: disable=unused-argument
```
Comment (quaat) [inline oteapi/strategies/download/sftp.py:2]: better to remove unused arguments than to disable to warning

File: oteapi/strategies/filter/crop_filter.py [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
     from typing import Any, Dict, Optional
 
-    from oteapi.models import FilterConfig
 
+class CropImageConfig(AttrDict):
+    """Configuration model for crop data."""
 
-class SessionUpdateCrop(SessionUpdate):
-    """Class for returning values from crop data."""
+    crop: Tuple[int, int, int, int] = Field(..., description="Box cropping parameters.")
```
Comment (quaat) [inline oteapi/strategies/filter/crop_filter.py:17]: nice

File: oteapi/strategies/transformation/celery_remote.py [hunk 3]
```diff
@@ -28,11 +38,12 @@ class SessionUpdateCelery(SessionUpdate):
...
 
 
-class CeleryConfig(BaseModel):
-    """Celery configuration."""
+class CeleryStrategyConfig(TransformationConfig):
+    """Celery strategy-specific configuration."""
 
-    task_name: str = Field(..., description="A task name.")
-    args: list = Field(..., description="List of arguments for the task.")
+    configuration: CeleryConfig = Field(
```
Comment (quaat) [inline oteapi/strategies/transformation/celery_remote.py:44]: clever

**Review by quaat — APPROVED** [review]

File: oteapi/strategies/download/sftp.py (hunk 1 above)
Comment (CasperWA) [reply oteapi/strategies/download/sftp.py:2]: Indeed. But I believe this is for the `session` parameter, which I believe we cannot remove? :/

**Pull request merged**